src/common/scheduler/scheduler.go

```diff
@@ -0,0 +1,258 @@
+package scheduler
+
+import "github.com/vmware/harbor/src/common/scheduler/policy"
+import "github.com/vmware/harbor/src/common/utils/log"
+
+import "errors"
+import "strings"
+import "fmt"
+import "reflect"
+import "time"
+
+const (
+	defaultQueueSize = 10
+
+	statSchedulePolicy   = "Schedule Policy"
+	statUnSchedulePolicy = "Unschedule Policy"
+	statTaskRun          = "Task Run"
+	statTaskComplete     = "Task Complete"
+	statTaskFail         = "Task Fail"
+)
+
+//StatItem is defined for the stat metrics.
+type StatItem struct {
+	//Metrics catalog
+	Type string
+
+	//The stat value
+	Value uint32
```

Comment:
Current situation is 1, not for other future possible scenarios. Let's keep it as it is.